src/include/duckdb/main/attached_database.hpp

```diff
@@ -31,16 +31,27 @@ enum class AttachedDatabaseType {
 	TEMP_DATABASE,
 };
 
+struct AttachOptions {
+	AttachOptions(AccessMode access_mode, const string &db_type);
+	AttachOptions(const unique_ptr<AttachInfo> &info, AccessMode access_mode_p);
+
+	AccessMode access_mode;
+	string db_type;
+	idx_t block_alloc_size;
+	string unrecognized_option;
```

Comment:
Shouldnt this be a `case_insensitive_map_t<Value> unrecognized_options;` ?

Reply:
I didn't touch this because, as far as I know, the last time I changed it, it blew up something on the MD side. 🤔 We are passing the original `AttachInfo` to the extension, no? So maybe it's just important to know that there was any unrecognized parameter?

```cpp
struct AttachInfo : public ParseInfo {
// ...
	//! Set of (key, value) options
	unordered_map<string, Value> options;
```

Reply:
Ah alright, nvm then 👍